### DISCUSSION TITLE
ItemsControl just display UserControl not ItemTemplate

### DISCUSSION THREAD

**Discussion by MintOTea** [AvaloniaUI/Avalonia#20415] [Q&A]
### Describe the bug

ItemsControl just display UserControl not ItemTemplate

### To Reproduce

I have a interface like this
![image](https://github.com/user-attachments/assets/cf780557-ea55-4273-9aa9-cec95df9a9fe)
and a UserControl extend this interface
![image](https://github.com/user-attachments/assets/fd37e829-115b-4f39-95fb-24d811f6d034)
![image](https://github.com/user-attachments/assets/12eb527e-28ca-462d-bd3a-e7ee53dd2063)

in ItemsControl i used ItemTemplate to display a Textblock with PaneTitle property which is "as" in interface
![image](https://github.com/user-attachments/assets/ebdf948e-e837-4a67-a547-ecaf74890a88)

It should display "as" in my ui, but it just display the content in the UserControl
![image](https://github.com/user-attachments/assets/3838bed3-b0a0-4096-aac1-e76c1b02af3e)

### Expected behavior

It should display "as" in my ui, but it just display the content in the UserControl
![image](https://github.com/user-attachments/assets/07ab1fa2-da95-44cb-939d-19d00c4b709b)

### Avalonia version

11.2.2

### OS

Windows

**Comment by rabbitism** [comment]
This is not a bug. 

https://github.com/AvaloniaUI/Avalonia/blob/master/src%2FAvalonia.Controls%2FItemsControl.cs#L497-L543

ItemsControl is designed to behave like this. As long as an element is a control, the container will display the control itself. 

If you dont need this feature, you should derive a new class, and override this method according to your needs. 

_Further more, UserControl as ItemsSource in general is valid but not a good practice_
